app/controllers/actors/AssetCancelProcessor.scala

```diff
@@ -2,43 +2,51 @@ package controllers
 package actors
 
 import scala.concurrent.duration._
-import models.{Asset, AssetLifecycle, MetaWrapper, Model, Status => AStatus}
+import models.{Asset, AssetLifecycle, MetaWrapper, Status => AStatus}
 import play.api.mvc.{AnyContent, Request}
 import util.plugins.SoftLayer
 import util.concurrent.BackgroundProcess
-import com.twitter.util.Await
+import scala.concurrent.{Await, ExecutionContext}
 
-case class AssetCancelProcessor(tag: String, userTimeout: Option[FiniteDuration] = None)(implicit req: Request[AnyContent])
+case class AssetCancelProcessor(tag: String, userTimeout: Option[FiniteDuration] = None)(implicit req: Request[AnyContent], ec : ExecutionContext)
   extends BackgroundProcess[Either[ResponseData,Long]]
 {
   override def defaultTimeout = 10 seconds
 
   val timeout = userTimeout.getOrElse(defaultTimeout)
+
+  private val noReason : Either[ResponseData,Long] = Left(Api.getErrorMessage("No reason specified for cancellation"))
+
+  private val softlayerDisabled : Either[ResponseData,Long] = Left(Api.getErrorMessage("SoftLayer plugin is not enabled"))
+
+  private val nonSoftlayerAsset : Either[ResponseData,Long] = Left(Api.getErrorMessage("Asset is not a softlayer asset"))
+
+  private val cancellationError : Either[ResponseData,Long] = Left(Api.getErrorMessage("There was an error cancelling this server"))
+
   def run(): Either[ResponseData,Long] = {
-    req.body.asFormUrlEncoded.flatMap(_.get("reason")).flatMap(_.headOption).map(_.trim).filter(_.size > 0).map { _reason =>
-      val reason = _reason.trim
+
+    val reasonOpt = req.body.asFormUrlEncoded.flatMap(_.get("reason")).flatMap(_.headOption).map(_.trim).filter(_.size > 0)
+
+    reasonOpt.fold(noReason){ r =>
+      val reason = r.trim
       Api.withAssetFromTag(tag) { asset =>
-        SoftLayer.pluginEnabled.map { plugin =>
-          plugin.softLayerId(asset) match {
-            case None =>
-              Left(Api.getErrorMessage("Asset is not a softlayer asset"))
-            case Some(n) =>
-              Await.result(plugin.cancelServer(n, reason)) match {
-                case 0L =>
-                  Left(Api.getErrorMessage("There was an error cancelling this server"))
-                case ticketId =>
-                  Asset.inTransaction {
-                    MetaWrapper.createMeta(asset, Map("CANCEL_TICKET" -> ticketId.toString))
-                    AssetLifecycle.updateAssetStatus(asset, AStatus.Cancelled, None, reason)
-                  }
-                  Await.result(plugin.setNote(n, "Cancelled: %s".format(reason)))
-                  Right(ticketId)
-              }
+        SoftLayer.pluginEnabled.fold(softlayerDisabled){ plugin =>
```

Comment:
This change looks a lot bigger than it is.  I was having trouble following it, so I decided to move some of the constants into separate declarations.  I also opted to use the "fold" function instead of map/getOrElse.